### PR TITLE
dotCMS/core#17983 Add PP filter dropdown to Angular PP dialog

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/view/components/_common/dot-push-publish-env-selector/dot-push-publish-env-selector.component.html [hunk 1]
```diff
@@ -3,17 +3,18 @@
...
-    [(ngModel)]="selectedEnvironments" 
-    optionLabel="name" 
-    defaultLabel="Select environment" 
-    (onChange)="valueChange($event, selectedEnvironments)">
+<p-multiSelect
+    (onChange)="valueChange($event, selectedEnvironments)"
+    [(ngModel)]="selectedEnvironments"
+    [options]="pushEnvironments$ | async"
+    appendTo="body"
+    defaultLabel="Select environment"
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-push-publish-env-selector/dot-push-publish-env-selector.component.html]: language key?

File: src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.html [hunk 2]
```diff
@@ -2,38 +2,129 @@
...
+            ></dot-field-validation-message>
         </div>
 
-        <div class="form__group push-publish-dialog__expire-date" *ngIf="form.get('pushActionSelected').value === 'expire' || form.get('pushActionSelected').value === 'publishexpire'">
-            <label class="form__label">{{ dotMessageService.get('contenttypes.content.push_publish.expire_date') }}: </label>
+        <div
+            class="form__group push-publish-dialog__expire-date"
+            *ngIf="
+                form.get('pushActionSelected').value === 'expire' ||
+                form.get('pushActionSelected').value === 'publishexpire'
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.html]: I think is better create a method for this or a Observable who listen by the form change, this way the logic is not repeat several times into the template

File: src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts [hunk 3]
```diff
@@ -42,34 +42,61 @@ export class DotPushPublishContentTypesDialogComponent implements OnInit, OnDest
...
+                        return {
+                            label: filter.title,
+                            value: filter.key
+                        };
+                    });
+
+                    const defaultFilterKey = filterOptions
+                        .filter((filter: DotPushPublishFilter) => {
+                            if (filter.default) {
+                                return filter;
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts]: what happend here if filter.default is false?

why not just:
```
return filter.default
```

return filter.default;

Comment (fmontes) [reply src/app/view/components/_common/dot-push-publish-dialog/dot-push-publish-dialog.component.ts]: Is not what I'm trying to do, I'm getting the object that have `default: true`

**Pull request merged**